homeassistant/components/webostv/media_player.py

```diff
@@ -65,131 +57,63 @@
-
-
-def setup_tv(
-    host, name, customize, config, timeout, hass, add_entities, turn_on_action
-):
-    """Set up a LG WebOS TV based on host parameter."""
-
-    client = WebOsClient(host, config, timeout)
-
-    if not client.is_registered():
-        if host in _CONFIGURING:
-            # Try to pair.
-            try:
-                client.register()
-            except PyLGTVPairException:
-                _LOGGER.warning("Connected to LG webOS TV %s but not paired", host)
-                return
-            except (OSError, ConnectionClosed, asyncio.TimeoutError):
-                _LOGGER.error("Unable to connect to host %s", host)
-                return
-        else:
-            # Not registered, request configuration.
-            _LOGGER.warning("LG webOS TV %s needs to be paired", host)
-            request_configuration(
-                host,
-                name,
-                customize,
-                config,
-                timeout,
-                hass,
-                add_entities,
-                turn_on_action,
-            )
-            return
+    client = hass.data[DOMAIN][host]["client"]
+    on_script = Script(hass, turn_on_action) if turn_on_action else None
 
-    # If we came here and configuring this host, mark as done.
-    if client.is_registered() and host in _CONFIGURING:
-        request_id = _CONFIGURING.pop(host)
-        configurator = hass.components.configurator
-        configurator.request_done(request_id)
+    device = LgWebOSMediaPlayerEntity(client, name, customize, on_script)
 
-    add_entities(
-        [LgWebOSDevice(host, name, customize, config, timeout, hass, turn_on_action)],
-        True,
-    )
+    hass.data[DOMAIN][host]["media_player"] = device
 
+    async_add_entities([device], update_before_add=False)
 
-def request_configuration(
-    host, name, customize, config, timeout, hass, add_entities, turn_on_action
-):
-    """Request configuration steps from the user."""
-    configurator = hass.components.configurator
 
-    # We got an error if this method is called while we are configuring
-    if host in _CONFIGURING:
-        configurator.notify_errors(
-            _CONFIGURING[host], "Failed to pair, please try again."
-        )
-        return
+def cmd(func):
+    """Catch command exceptions."""
 
-    def lgtv_configuration_callback(data):
-        """Handle actions when configuration callback is called."""
-        setup_tv(
-            host, name, customize, config, timeout, hass, add_entities, turn_on_action
-        )
+    @wraps(func)
+    async def wrapper(obj, *args, **kwargs):
+        """Wrap all command methods."""
+        try:
+            await func(obj, *args, **kwargs)
+        except (
+            asyncio.TimeoutError,
+            asyncio.CancelledError,
+            PyLGTVCmdException,
+        ) as exc:
+            # If TV is off, we expect calls to fail.
+            if obj.state == STATE_OFF:
+                log_function = _LOGGER.info
```

Comment:
We can use the `log` method of the logger to pass the logging level as an argument.
```py
_LOGGER.log(level, ...)
```

https://docs.python.org/3/library/logging.html#logging.Logger.log